### ISSUE TITLE
Outline Compatability to more recent minio version - after upgrade.

### ISSUE THREAD

**Issue by UKFr-DIZ** [outline/outline#6505]
### TLDR
We upgraded minio without changing any other configs in outline. The inplace minio upgrade worked fine and outline user access, was tested against the minio gui, being able to access all files in the relevant buckets. After upgrade of minio the outline minio user cannot access anything in the bucket anymore. Even when the bucket policies are set to public. 

### Context 
Hi, 

we have upgraded our minio s3 storage from `minio_20230210184839` to `2024-01-31T20` (~1 year difference). The minio instance is using Keycloak as an IdP however that shouldn't be an issue as far as I know. I've created a access key pair in minio, and use those credentials in the outline configuration in order to integrate s3 into the outline instance. Minio runs beind a apache revproxy, but that shouldn't explain the error mesage we get. 

Since the upgrade, using the minio API results in HTTP 403 Forbidden error messages for the private buckets / attachements which are stored in s3. 
We haven't changed a single thing in the outline configuration and we suspect that the AWS SDK S3 Client isn't compatible anymore with current minio instance anymore. However maybe we are wrong in that. We have testet, that the access key login via the minio console works, and it can read all bucket data. We actually even increased the IAM policy by add `s3:*` to its policy. 

### **Error messages in details:** 

URL Call: 
`https://<s3-server>:9000/wiki/uploads/5e846089-f7c9-4f87-9dfa-b5af6b54c2cb/53af5eef-90a4-4576-aaed-7fb19e43b2b5/picture.png`
Type: `GET`
Payload: 
```
X-Amz-Algorithm: AWS4-HMAC-SHA256
X-Amz-Credential: <AWS-ACCESS-ID-Short>/20240206/us-west-2/s3/aws4_request
X-Amz-Date: 20240206T164521Z
X-Amz-Expires: 60
X-Amz-Signature: bcac0399b79e37137633a7a87e5aa9e615ede92951ba532687b83a8cc2849ec8
X-Amz-SignedHeaders: host
response-content-disposition: attachment
```

Error msg: 
~~~
<Error>
<Code>AccessDenied</Code>
<Message>Access Denied.</Message>
<Key>uploads/5e846089-f7c9-4f87-9dfa-b5af6b54c2cb/53af5eef-90a4-4576-aaed-7fb19e43b2b5/picture.png</Key>
<BucketName>wiki</BucketName>
<Resource>/wiki/uploads/5e846089-f7c9-4f87-9dfa-b5af6b54c2cb/53af5eef-90a4-4576-aaed-7fb19e43b2b5/picture.png</Resource>
<Region>us-west-2</Region>
<RequestId>17B1540041C9F88A</RequestId>
<HostId>7987905dee74cdeb212432486a178e511309594cee7cb75f892cd53e35f09ea4</HostId>
</Error>
~~~

### **To Reproduce**
 - Have outline setup with s3 Minio store in version `minio_20230210184839`. minio is secured via Keycloak
 - Create bucket + "admin" policy in minio, for the bucket and assign it to the Keycloak user (e.g. policy attribute + claim)
 - Create Access keypair, which are used as credentials in outline
 - Create a document in outline and upload pictures
 - Upgrade minio s3 to `2024-01-31`
 - try access documents in outline, with the uploaded attachements / pictures

#### relevant configurations: 

`.env` outline:
~~~
....
AWS_S3_ACCELERATE_URL=
AWS_REGION=us-west-2
AWS_S3_UPLOAD_BUCKET_URL=https://<s3-server>:9000
AWS_S3_UPLOAD_BUCKET_NAME=wiki
AWS_S3_FORCE_PATH_STYLE=true
AWS_S3_ACL=private

FILE_STORAGE=s3

#FILE_STORAGE_LOCAL_ROOT_DIR=/var/lib/outline/data

# Maximum allowed size for the uploaded attachment.
FILE_STORAGE_UPLOAD_MAX_SIZE=26214400
...
~~~

### **Expected behavior**
The attachements.redirect method work properly and the images are displayed in the document.

### **Outline (please complete the following information):**
- Install: self-hosted
- Version: latest 0.74.0

### **Desktop (please complete the following information):**
 - OS: Widnows 10
 - Browser Edge

Thanks for any support! :) We can provide more details ofc.

**Comment by UKFr-DIZ** [comment]
We have found out that minio had a bug in version `2024-01-31T20`. 
It has been fixed with the release afterwards: https://github.com/minio/minio/releases/tag/RELEASE.2024-02-04T22-36-13Z

Apparently the version we upgraded had a bug that access key authentication didn't work.

> Fixes a regression in service accounts/access keys for the root-user stopped working in the last release,
> a side-affect of a security bug-fix has been addressed.
